test/should_pass/record_wildcard.erl

```diff
@@ -15,3 +15,10 @@ f() ->
 g() ->
     #rec{ apa = 1
         , _   = true }.
+
+%% wildcard in pattern matching
+-spec h(#rec{}) -> boolean().
+h(#rec{apa = 1, _ = true}) ->
```

Comment:
this is something the compiler allows, but I dont think it's quite useful or used. Should we allow this at all or rather throw a warning (similar to lazy-andalso) discouraging the usage?

Reply:
It's quite much used in some code that I've seen, so I think we should allow it.

Reply:
E.g. when writing patterns for mnesia. I wonder how we could support that, but that's for the future...

Reply:
do you see it in record creation or record patterns? do you see it with any other value than the underscore atom `'_'` or `'undefined'`? it's totally fine with creating match-specs for ets/mnesia. (`ets:fun2ms(fun(#rec{apa = 1, _ = undefined}) -> ...` is also converted to record creation and not a pattern by the `ms_transform` parse trans). Given that could you give a real-world example of using wildcard in record pattern?

I'm fine with supporting this...also I realise this is the wrong PR, I just found this case when reviewing my code changes.

Reply:
I've never seen it in a pattern. Only in expressions. I've seen it with `_ = false` in a record where all fields are booleans.